pyomo/contrib/trustregion/GeometryGenerator.py

```diff
@@ -11,7 +11,10 @@
 import logging
 
 # This is an auto geometry generator for quadratic ROM
-import numpy as np
+from pyomo.common.dependencies import numpy_available
+if numpy_available:
+    import numpy as np
```

Comment:
The simpler thing to do here is just:
```python
from pyomo.common.dependencies import numpy as np
```
(numpy will be automatically imported the first time it is used).  That also avoids importing the dependency unless someone is actually using trustregion.